### ISSUE TITLE
 Packery appended items twitching

### ISSUE THREAD

**Issue by matyushen** [metafizzy/packery#287]
I'm trying to create the ajax load more posts functionality for a WordPress theme. Like shown in this example http://codepen.io/desandro/pen/HFbia. Looks pretty clean. But I'm facing a problem with appending new posts to Packery. When I trigger loading new items, they all first appear in the top left corner of the Packery container, and then they quickly fall into place.

Here's the js:

    jQuery(document).ready(function($) {
	 
		// The number of the next page to load (/page/x/).
		var pageNum = parseInt(theme.startPage) + 1;
	 
		// The maximum number of pages the current query can return.
		var max = parseInt(theme.maxPages);
	 
		// The link of the next page of posts.
		var nextLink = theme.nextLink;

		// Button text
		var btnText = theme.btnText;

		// Loading button text
		var btnTextLoading = theme.btnTextLoading;

		// No more posts to load message
		var btnTextNoMore = theme.btnTextNoMore;
		

		// Packery container
		var loop = $('.site-loop');

		// Fire Packery
		loop.imagesLoaded( function() {
			loop.packery({
				itemSelector: '.post',
				gutter: '.site-loop-gutter',
				percentPosition: true,
				transitionDuration: '0.2s'
			});
		});
	 

		if( pageNum <= max ) {

			// Insert the "More Posts" link.
			$('#loop-navigation').append('<div id="theme-load-posts" class="ajax-navigation"><a href="">' + btnText + '</a></div>');
	 
			// Remove the standard WordPress navigation.
			$('.navigation').remove();
		}

		// Show new posts when the link is clicked.
		$('#theme-load-posts a').click( function() {
	        
			if( pageNum <= max ) {

				// Loading progress message
				$('#theme-load-posts a').text( btnTextLoading );

				// 
				$.get( nextLink , function( html ) {

					// New posts to show
					var elems = $(html).find('.site-loop').html();
					var $elems = $(elems);

					if (elems) {

						// append elements to container
						loop.packery().append($elems);

						loop.imagesLoaded( function() {
							// add and lay out newly appended elements
							loop.packery('appended', $elems);
						});

					};

	            });

	            pageNum++;
	                
	            nextLink = nextLink.replace(/\/page\/[0-9]*/, '/page/'+ pageNum);

	            loop.on( 'layoutComplete',
	            	function( event, laidOutItems ) {

	            		if(pageNum <= max) {
	            			$('#theme-load-posts a').text( btnText );
	            		} else {
	            			$('#theme-load-posts').text(btnTextNoMore);
			            }
		            }
	             );
	 
			} else {

				$('#theme-load-posts').text(btnTextNoMore);

			}

			return false;

		});

	});

Would appreciate any help!

**Comment by desandro** [comment]
I'm sorry to see you're having trouble with Packery. Could you provide a **reduced test case**? See [Submitting Issues in the contributing guidelines](https://github.com/metafizzy/packery/blob/master/contributing.md#submitting-issues).

**Comment by matyushen** [comment]
Thanks for your answer. It's hard to provide a test case since I'm developing WordPress locally. But I totally understand that it's hard to help without it. I was just hoping that I missed something obvious. 

I managed to solve the problem, though, I believe. Turned out that I need to wait before images are loaded in the newly created bunch of posts before appending them:

    $(elems).imagesLoaded( function() {
        loop.packery().append($elems);
        loop.packery('appended', $elems);
    });

That's it, no more items twitching all over the screen. Hope this will help somebody.
